web4/treasury-web4/src/lib.rs

```diff
@@ -64,6 +90,61 @@ impl Contract {
         }
     }
 
+    #[private]
+    pub fn set_social_metadata(
+        &mut self,
+        social_db_account_id: Option<String>,
+        name: Option<String>,
+        description: Option<String>,
+        ipfs_cid: Option<String>,
+    ) -> Promise {
+        self.internal_set_social_metadata(social_db_account_id, name, description, ipfs_cid)
+    }
+
+    fn internal_set_social_metadata(
+        &mut self,
+        social_db_account_id: Option<String>,
+        name: Option<String>,
+        description: Option<String>,
+        ipfs_cid: Option<String>,
+    ) -> Promise {
+        let social_db_account_id =
+            social_db_account_id.unwrap_or_else(|| "social.near".to_string());
+        let name = name.unwrap_or_else(|| "NEAR treasury".to_string());
+        let description = description
+            .unwrap_or_else(|| format!("NEAR treasury for {}", env::current_account_id().as_str()));
+        let ipfs_cid = ipfs_cid.unwrap_or_else(|| {
+            "bafkreiboarigt5w26y5jyxyl4au7r2dl76o5lrm2jqjgqpooakck5xsojq".to_string()
+        });
+
+        let args = near_sdk::serde_json::json!(
+        {"data":
+            {env::current_account_id():
+                {"widget":
```

Comment:
nit: It is not conventional formatting for JSON, it would be great to format it:

```rust
        let args = near_sdk::serde_json::json!({
            "data": {
                env::current_account_id(): {
                    "widget":
```